extension/src/paymentHandler/payment-handler.js

```diff
@@ -20,22 +23,72 @@ const PAYMENT_METHOD_TYPE_KLARNA_METHODS = [
   'klarna_account',
 ]
 
-async function handlePayment(paymentObject) {
+function _validatePaymentRequest(paymentObject, authToken) {
```

Comment:
And do not know if you mind but I personally appreciate when functions are defined within the file in the same order as they are being used (from top to bottom) and one does not have to jump back and forth when looking through the logic.

Reply:
Rearrange the order. Put the main method at the top, and put _validatePaymentRequest() after _isAdyenPayment()